File: day06/day6.cpp
#include <iostream>

using namespace std;
#define D 256

long long get_total(int first) {
    long long total = 1;
    long long modCount[D][7] = {0};
    modCount[0][first] = 1;
    for (int i = 1; i < D; i++) {
        int parent = i % 7;
        int child = (i + 2) % 7;
        for (int j = 0; j < 7; j++) {
            modCount[i][j] += modCount[i - 1][j];
        }
        total += modCount[i][parent];
        if (i + 3 < D) {
            modCount[i + 3][child] += modCount[i][parent];
        }
    }
    return total;
}

int main() {
    long long modTotal[5] = {0};
    for (int i = 0; i < 5 ; i++) {
        modTotal[i] = get_total(i + 1);
    }
    long long total = 0;

    int fish;
    while (cin >> fish) {
        total += modTotal[fish - 1];
        cin.ignore(1);
    }

    cout << total << endl;
}

File: day06/day6.py
# REPLACE 256 to 80 FOR PART 1 OF DAY 6

def get_total(first):
    total = 1
    fish = [[0] * 7 for i in range(256)]
    fish[0][first] = 1
    for i in range(1, 256):
        parent = i % 7
        baby = (i + 2) % 7
        for j in range(7):
            fish[i][j] += fish[i - 1][j]
        total += fish[i][parent]
        if i + 3 < 256:
            fish[i + 3][baby] += fish[i][parent]
    return total

ending_fish = []
for i in range(5):
    ending_fish.append(get_total(i + 1))

total = 0

with open("input.txt", "r") as f:
    starting_fish = list(map(int, f.readline().strip().split(",")))

for fish in starting_fish:
    total += ending_fish[fish - 1]


print(total)

